mttl/models/modifiers/sparse_mask.py

```diff
@@ -15,7 +15,63 @@
 
 from mttl.models.modifiers.base import Modifier, ModifierConfig
 from mttl.utils import logger
+from collections import OrderedDict
 
+def hook_factory(keep_mask):
+    """
+    The hook function can't be defined directly here because of Python's
+    late binding which would result in all hooks getting the very last
+    mask! Getting it through another function forces early binding.
+    """
+    def hook(grads):
+        return grads * keep_mask
+    return hook
+
+def load_mask(f_name):
```

Comment:
Is there a way we can avoid needing these functions `save_mask` and `load_mask`?

weight_mask should be included in state_dict by now so can be reloaded from checkpoints?